tests/test_worktree_cleanup.sh

```
@@ -0,0 +1,379 @@
+EXT_DIR="$WORKDIR/external-canary"
+mkdir -p "$EXT_DIR"
+CANARY_FILE="$EXT_DIR/do-not-delete.txt"
+echo "this file MUST survive symlink-escape attempts" > "$CANARY_FILE"
+
+# Set up the symlink under .claude/worktrees/.
+ln -s "$EXT_DIR" "$MAIN/.claude/worktrees/agent-symlink-escape"
+
+# Run --apply --orphan-clean and capture output.
+set +e
+OUT_ESC=$(PATH="$STUB_DIR:$PATH" bash "$HELPER" --no-color --apply --orphan-clean 2>&1)
+RC_ESC=$?
```

Shellcheck (0.11.0)

[warning] RC_ESC appears unused. Verify use (or export if used externally).

(SC2034)